connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/HerderRequestHandler.java

```diff
@@ -41,18 +41,11 @@ public class HerderRequestHandler {
 
     private final RestClient restClient;
 
-    private volatile long requestTimeoutMs;
+    private final RestRequestTimeout requestTimeout;
 
-    public HerderRequestHandler(RestClient restClient, long requestTimeoutMs) {
+    public HerderRequestHandler(RestClient restClient, RestRequestTimeout requestTimeout) {
         this.restClient = restClient;
-        this.requestTimeoutMs = requestTimeoutMs;
-    }
-
-    public void requestTimeoutMs(long requestTimeoutMs) {
-        if (requestTimeoutMs < 1) {
-            throw new IllegalArgumentException("REST request timeout must be positive");
-        }
```

Comment:
I am not sure if this validation was needed in the past as well, but now I don't see it being present. Do you think it's needed?

Reply:
I removed it for two reasons:

1. This code path is only ever used in testing code
2. It's unclear that the validation would be safer than allowing negative values. At least on my machine, it doesn't cause any issues with, e.g., `ConvertingFutureCallback::get`.